contrib/cirrus/lib.sh

```diff
@@ -3,293 +3,188 @@
 # Library of common, shared utility functions.  This file is intended
 # to be sourced by other scripts, not called directly.
 
+# BEGIN Global export of all variables
+set -a
+
 # Global details persist here
-source /etc/environment  # not always loaded under all circumstances
+source /etc/environment || touch /etc/environment  # needed on Mac OSX
+
+# Under some contexts these values are not set, make sure they are.
+USER="$(whoami)"
+HOME="$(getent passwd $USER | cut -d : -f 6)"
+[[ -n "$UID" ]] || \
+    UID=$(getent passwd $USER | cut -d : -f 3)
+GID=$(getent passwd $USER | cut -d : -f 4)
```

Comment:
Whew. I see that this is simply a move of already-existing code, but *shudder*. Some day (not this iteration!) I'd like to understand this better, see which systems it fails on, use `${...:-...}` to set only if missing, and perhaps use `id` instead of getent. ITM if you need to repush for some other reason, a quick FIXME here might be nice.

Reply:
I think this was for Ubuntu, during initial implementations, though I added the comment later on.  In hind-sight, I think I understand why this was needed and am willing to try removing it.

Reply:
While attempting to remove this, I re-discovered why it's necessary and added a helpful comment.  The primary reason we need it is for the tasks which run inside a container.  In those cases, things like `$HOME` and `$USER` are ***not*** set - perhaps due to the way cirrus executes containers.